src/fix.ts

```diff
@@ -0,0 +1,98 @@
+import { OAuth2Client } from 'google-auth-library';
+import fs from 'fs';
+import readline from 'readline';
+import {
+  REDIRECT_URL,
+  JSON_FILE_PATH,
+  googleClientID,
+  googleClientSecret,
+  getCredentials,
+  getCredentialsFromJSON,
+  getEvents,
+  deleteEvents,
+  createEvent,
+} from './util/googleApiUtility';
+import {
+  displayDateTimeRange,
+  dividedDateTimeRange,
+  groupEventsByDate,
+  isInRange,
+} from './util/dateTimeUtility';
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getResponse = async (question: string): Promise<string> => {
+  return new Promise((resolve) => {
+    rl.question(question, (response) => {
+      const lowerCaseResponse = response.toLowerCase();
+      if (
+        lowerCaseResponse === 'y' ||
+        lowerCaseResponse === 'yes' ||
+        lowerCaseResponse === 'n' ||
+        lowerCaseResponse === 'no'
+      ) {
+        resolve(lowerCaseResponse);
+      } else {
+        resolve(getResponse(question));
+      }
+    });
+  });
+};
+
+const fixCommandHandler = async () => {
+  const oauth2Client = new OAuth2Client(
+    googleClientID,
+    googleClientSecret,
+    REDIRECT_URL
+  );
+
+  const credentials = fs.existsSync(JSON_FILE_PATH)
+    ? getCredentialsFromJSON(JSON_FILE_PATH)
+    : await getCredentials(oauth2Client);
+  if (credentials) {
+    oauth2Client.setCredentials(credentials);
+  }
+
+  const dateTimeRange = process.argv[2];
+  const afterEventName = process.argv[3];
+  const beforeEventName = process.argv[4];
+
+  try {
+    const beforeEvents = await getEvents(oauth2Client, beforeEventName);
+    if (beforeEvents) {
+      const groupedEvents = groupEventsByDate(beforeEvents);
+      await displayDateTimeRange(groupedEvents);
+      const [startDateTime, endDateTime] = dividedDateTimeRange(dateTimeRange);
+      if (isInRange(startDateTime, endDateTime, beforeEvents)) {
+        console.log('Are you sure to remove schedules?');
+        const response = await getResponse(
+          `And add this?\n${dateTimeRange}\n(y/n) > `
+        );
+        const lowerCaseResolve = response.toLowerCase();
+        if (lowerCaseResolve === 'y' || lowerCaseResolve === 'yes') {
```

Comment:
これって一件づつ確認出ますか?

3件登録されている場合
2件=> 削除
1件=> 変更
にして、確認は1回で済ませたいです

Reply:
削除の確認は1度しか表示されないようになっています。
y / yes / n / no 以外のキーが押される度に、削除の確認が表示されるようになっています。

Reply:
こんな感じです（動画ではEnterを連打してます）
https://github.com/mk-system/timehunt/assets/61813198/cab1f5dc-ad32-4eb9-b3b0-1182d6403150

Reply:
※参考（Enter連打した場合）
https://github.com/mk-system/timehunt/assets/61813198/aa03b01b-0abe-4d95-bb0d-7e4310245068

Reply:
全カレンダーの確認がでて、
一回yをおしたら、消すのと登録するのを全部やって欲しいです

Reply:
申し訳ありません。要件を確認させて頂きたいのですが、
yを押下した場合、第三引数で指定したイベント名に一致するイベントを全て取得・削除、第一・二引数で指定した日時とイベント名で登録しますが、この動きではない感じですか？
もし違うのであれば、もう少し詳細に書いて頂けますと助かります。

Reply:
> yを押下した場合、第三引数で指定したイベント名に一致するイベントを全て取得・削除、第一・二引数で指定した日時とイベント名で登録します

個々までは日本語としてはあっています
y 連打は避けたいです。y or n は1回しか押したくないです。

上ここまでが要件になります。
仕様も書いちゃう場合

```
$ pnpm run fix xx打ち合わせ 2023-12-14 11:00-12:00
It will remove these schedules
2023-12-13 10:00-19:00
2023-12-14 10:00-19:00
And it will change this schedule.
2023-12-15 10:00-19:00 to 10:00-12:00
Are you sure?
> y
done
```

こんな感じです。

Reply:
現状の実装でも y or nは1度押すだけで大丈夫です（動画ではy / n以外のキーが押された場合の動きも含めています）。

Reply:
すみません動かしたらイメージ通りでした。